plugins/auth-backend/src/service/OidcService.ts

```diff
@@ -89,4 +108,404 @@ export class OidcService {
     }
     return await this.userInfo.getUserInfo(userEntityRef);
   }
+
+  public async registerClient(opts: {
+    responseTypes?: string[];
+    grantTypes?: string[];
+    clientName: string;
+    redirectUris?: string[];
+    scope?: string;
+  }) {
+    const generatedClientId = crypto.randomUUID();
+    const generatedClientSecret = crypto.randomUUID();
+
+    const allowedRedirectUriPatterns = this.config.getOptionalStringArray(
+      'auth.experimentalDynamicClientRegistration.allowedRedirectUriPatterns',
+    );
+
+    if (allowedRedirectUriPatterns) {
+      for (const redirectUri of opts.redirectUris ?? []) {
+        if (
+          !allowedRedirectUriPatterns.some(pattern =>
+            new RegExp(pattern).test(redirectUri),
```

Comment:
i wonder if full regex (you know, easy to forget to escape dots etc which could have true security implications) is overkill and minimatch / globby / whatnot would work out better?